File: leetcode/2865.美丽塔 I.go
func maximumSumOfHeights(maxHeights []int) int64 {
    n := len(maxHeights)
    res := int64(0)
    for i := 0; i < n; i ++ {
        pre, psum := maxHeights[i], int64(maxHeights[i])
        for j := i - 1; j >= 0; j -- {
            pre = min(pre, maxHeights[j])
            psum += int64(pre)
        }
        suf := maxHeights[i]
        for j := i + 1; j < n; j++ {
            suf = min(suf, maxHeights[j])
            psum += int64(suf)
        }
        res = max(res, psum)
    }
    return res
}

File: leetcode/2865.美丽塔 I.py
class Solution:
    def maximumSumOfHeights(self, mh: List[int]) -> int:
        # 枚举峰值
        n = len(mh)
        ans = 0
        for t in range(n):
            tot = mh[t]
            # print(tot)
            i, j = t-1, t+1
            pre = tot
            while i >= 0:
                cur = min(pre, mh[i])
                # print(i, cur)
                tot += cur
                pre = cur
                i -= 1
            pre = tot
            while j < n:
                cur = min(pre, mh[j])
                # print(j, cur)
                tot += cur
                pre = cur
                j += 1
            ans = max(ans, tot)
        return ans